daemon/src/model.rs

```diff
@@ -692,17 +680,155 @@ impl str::FromStr for OpeningFee {
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         let amount_sat: u64 = s.parse()?;
-        Ok(OpeningFee { amount_sat })
+        Ok(OpeningFee {
+            fee: Amount::from_sat(amount_sat),
+        })
     }
 }
 
+impl_sqlx_type_display_from_str!(OpeningFee);
+
 impl Default for OpeningFee {
     fn default() -> Self {
-        Self::new(Amount::ZERO)
+        Self { fee: Amount::ZERO }
     }
 }
 
-impl_sqlx_type_display_from_str!(OpeningFee);
+/// Funding fee as defined by position and rate
+///
+/// Position and rate define if the funding rate is to be added or subtracted when summing up the
+/// fees.
+#[derive(Copy, Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct FundingFee {
+    #[serde(with = "bdk::bitcoin::util::amount::serde::as_sat")]
+    fee: Amount,
+    rate: FundingRate,
+}
+
+impl FundingFee {
+    pub fn new(fee: Amount, rate: FundingRate) -> Self {
+        Self { fee, rate }
+    }
+}
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum FeeFlow {
+    LongPaysShort(Amount),
+    ShortPaysLong(Amount),
+    Nein,
+}
+
+/// Our own accumulated fees
+///
+/// The balance being positive means we owe this amount to the other party.
+/// The balance being negative means that the other party owes this amount to us.
+/// The counterparty fee-account balance is always the inverse of the balance.
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct FeeAccount {
+    balance: SignedAmount,
+    position: Position,
+    role: Role,
+}
+
+impl FeeAccount {
+    pub fn new(position: Position, role: Role) -> Self {
+        Self {
+            position,
+            role,
+            balance: SignedAmount::ZERO,
+        }
+    }
+
+    pub fn settle(&self) -> FeeFlow {
+        let absolute = self.balance.as_sat().unsigned_abs();
+        let absolute = Amount::from_sat(absolute);
+
+        if self.balance == SignedAmount::ZERO {
+            FeeFlow::Nein
+        } else if (self.position == Position::Long && self.balance.is_positive())
+            || (self.position == Position::Short && self.balance.is_negative())
+        {
+            FeeFlow::LongPaysShort(absolute)
+        } else {
+            FeeFlow::ShortPaysLong(absolute)
+        }
+    }
+
+    pub fn balance(&self) -> SignedAmount {
+        self.balance
+    }
+
+    pub fn add_opening_fee(self, opening_fee: OpeningFee) -> Self {
+        let fee: i64 = opening_fee
+            .fee
+            .as_sat()
+            .try_into()
+            .expect("not to overflow");
+
+        let signed_fee = match self.role {
+            Role::Maker => -fee,
+            Role::Taker => fee,
+        };
+
+        let signed_fee = SignedAmount::from_sat(signed_fee);
+        let sum = self.balance + signed_fee;
+
+        Self {
+            balance: sum,
+            position: self.position,
+            role: self.role,
+        }
+    }
+
+    pub fn add_funding_fee(self, funding_fee: FundingFee) -> Self {
```

Comment:
Nit: I think this is a good place to add some domain knowledge about funding fees!